### PR TITLE
Add course level redirects when access is unauthorized

### PR DESCRIPTION
**Pull request by jpaulcrosby**

### CODE REVIEW

File: admin/class-wp-classroom-admin.php [hunk 1]
```diff
@@ -223,7 +223,8 @@ public function wp_classroom_taxonomy() {
 	 		'not_found'                  => __( 'Not Found', 'wp-classroom' ),
 	 		'no_terms'                   => __( 'No Courses', 'wp-classroom' ),
 	 		'items_list'                 => __( 'Course list', 'wp-classroom' ),
-	 		'items_list_navigation'      => __( 'Course list navigation', 'wp-classroom' ),
+			'items_list_navigation'      => __( 'Course list navigation', 'wp-classroom' ),
+			'course_redirect'            => __( 'No access to course', 'wp-classroom')
```
Comment (greghunt) [inline admin/class-wp-classroom-admin.php]: @jpaulcrosby what's this for? It's not an option in the expected labels:
https://developer.wordpress.org/reference/functions/get_taxonomy_labels/

File: admin/class-wp-classroom-admin.php [hunk 2]
```diff
@@ -306,7 +308,14 @@ function associate_course_teachers() {
 			'select_all_button' => false,
 			'options' => $this->teacherOptions(),
 		) );
-	}
+
+		$cmb_term->add_field( array(
+			'name'     => __( 'Course Redirect', $this->plugin_name ),
+			'id'       => $prefix . 'course_redirect',
+			'type'    => 'text',
+			'taxonomies'  => array( 'wp_course' ),
```
Comment (greghunt) [inline admin/class-wp-classroom-admin.php:316]: this should use term_meta as well, shouldn't it?

File: includes/class-wp-classroom-users.php [hunk 3]
```diff
@@ -149,10 +149,14 @@ public function redirect()
     {
         global $post;
         $url = wp_login_url();
-
+        $all_terms = get_the_terms($post->ID, get_taxonomies())[0];
+        $course_id = $all_terms->parent ? $all_terms->parent : $all_terms->term_id;
```
Comment (greghunt) [inline includes/class-wp-classroom-users.php]: Why are we using get_taxonomies() ? This is only applicable to `wp_course`. Also, we're only checking the first term for some reason. There should probably be a loop somewhere to check presence of redirect on each term this class has.

**Review by greghunt — CHANGES_REQUESTED** [review]

File: admin/class-wp-classroom-admin.php (hunk 1 above)
Comment (jpaulcrosby) [reply admin/class-wp-classroom-admin.php]: You are totally right. We can delete this. Ill resubmit on this one.

**Review by greghunt — APPROVED** [review]

**Pull request merged**